pkg/cmd/server/start/master_args.go

```diff
@@ -71,16 +71,25 @@ func NewDefaultMasterArgs() *MasterArgs {
 		MasterPublicAddr: flagtypes.Addr{Value: "localhost:8443", DefaultScheme: "https", DefaultPort: 8443, AllowPrefix: true}.Default(),
 		DNSBindAddr:      flagtypes.Addr{Value: "0.0.0.0:53", DefaultScheme: "http", DefaultPort: 53, AllowPrefix: true}.Default(),
 
+		ConfigDir: &util.StringFlag{},
```

Comment:
> so you can't control where we generate stuff while just starting? setting the config dir implicitly tells it to write config and exit?

Yes to both.



> it's weird that we write to the default value of --write-config when you don't specify it

The other way around makes more sense though.  `configDir` always exists.  A user can set its value using the `--write-config` flag.